src/algorithms/reco/JetReconstruction.cc

```diff
@@ -83,16 +82,33 @@ namespace eicrecon {
     }
     m_log->trace("  Number of particles: {}", particles.size());
 
-    // Choose jet and area definitions
-    JetDefinition jet_def(m_mapJetAlgo[m_cfg.jetAlgo], m_cfg.rJet, m_mapRecombScheme[m_cfg.recombScheme]);
-    AreaDefinition area_def(m_mapAreaType[m_cfg.areaType], GhostedAreaSpec(m_cfg.ghostMaxRap, m_cfg.numGhostRepeat, m_cfg.ghostArea));
+    // Choose jet definition based on no. of parameters
+    const int nParams = get_num_params();
+    switch (nParams) {
+
+      case 0:
+        m_jet_def = std::make_unique<JetDefinition>( m_mapJetAlgo[m_cfg.jetAlgo], m_mapRecombScheme[m_cfg.recombScheme] );
+        break;
+
+      case 2:
+        m_jet_def = std::make_unique<JetDefinition>( m_mapJetAlgo[m_cfg.jetAlgo], m_cfg.rJet, m_cfg.pJet, m_mapRecombScheme[m_cfg.recombScheme] );
+        break;
+
+      default:
+        m_jet_def = std::make_unique<JetDefinition>( m_mapJetAlgo[m_cfg.jetAlgo], m_cfg.rJet, m_mapRecombScheme[m_cfg.recombScheme] );
+        break;
+
+    }  // end switch (jet algorithm)
+
+    // Define jet area
+    m_area_def = std::make_unique<AreaDefinition>( m_mapAreaType[m_cfg.areaType], GhostedAreaSpec(m_cfg.ghostMaxRap, m_cfg.numGhostRepeat, m_cfg.ghostArea) );
```

Comment:
Can this configuration not be called in the initialization?

Reply:
I agree with @simonge that it would seem to make more sense to move creating the jet and area definitions to the initialization as in principle this only needs to be done once.

Reply:
Oh! Good point! I'll definitely move this into `init()`...